pandas/_libs/algos.pyx

```diff
@@ -451,11 +448,8 @@ def nancorr_spearman(ndarray[float64_t, ndim=2] mat, Py_ssize_t minp=1) -> ndarr
                             with gil:
                                 # We need to slice back to nobs because rank_1d will
                                 # require arrays of nobs length
-                                labels_nobs = np.zeros(nobs, dtype=np.int64)
-                                rankedx = rank_1d(np.array(maskedx)[:nobs],
-                                                  labels=labels_nobs)
-                                rankedy = rank_1d(np.array(maskedy)[:nobs],
-                                                  labels=labels_nobs)
+                                rankedx = rank_1d(np.array(maskedx)[:nobs])
+                                rankedy = rank_1d(np.array(maskedy)[:nobs])
```

Comment:
not sure if it makes a difference, but np.array -> np.asarray?

Reply:
good catch, was an oversight no reason to copy here